x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+        searchAfterSortId,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(searchAfterResult).toEqual(sampleDocSearchResultsWithSortId);
+    });
+    test('if singleSearchAfter throws error', async () => {
+      const searchAfterSortId = '1234567891111';
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const expectedSearchAfterQuery = buildEventsSearchQuery({
+        index: sampleParams.index,
+        from: sampleParams.from,
+        to: sampleParams.to,
+        filter: sampleParams.filter,
+        size: sampleParams.size ? sampleParams.size : 1,
+        searchAfterSortId,
+        maxDocs: undefined,
+      });
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: SearchParams) => {
+          expect(action).toEqual('search');
+          expect(params.index).toEqual(sampleParams.index);
+          expect(params).toEqual(expectedSearchAfterQuery);
+          throw Error('Fake Error');
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      try {
+        await singleSearchAfter(searchAfterSortId, sampleParams, mockService, mockLogger);
+      } catch (exc) {
+        expect(mockLogger.error).toHaveBeenCalledTimes(1);
+        expect(exc.message).toEqual('Fake Error');
+      }
+    });
+  });
+  describe('searchAfterAndBulkIndex', () => {
+    test('if successful with empty search results', async () => {
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const mockService: AlertServices = {
+        callCluster: jest.fn(),
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const result = await searchAfterAndBulkIndex(
+        sampleEmptyDocSearchResults,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockService.callCluster).toHaveBeenCalledTimes(0);
+      expect(result).toEqual(true);
+    });
+    test('if one successful iteration of searchAfterAndBulkIndex', async () => {
+      const searchAfterSortId = '1234567891111';
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const expectedSearchAfterQuery = buildEventsSearchQuery({
+        index: sampleParams.index,
+        from: sampleParams.from,
+        to: sampleParams.to,
+        filter: sampleParams.filter,
+        size: sampleParams.size ? sampleParams.size : 1,
+        searchAfterSortId,
+        maxDocs: undefined,
+      });
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: SearchParams) => {
+          if (action === 'bulk') {
+            expect(action).toEqual('bulk');
+            expect(params.index).toEqual('.siem-signals-10-01-2019');
```

Comment:
These checks for SIEM signals do fail if I have a custom SIEM dev signals configured via my environment. We will be fixing/changing/removing the env variables but for now I would add a simple function for these tests which does:

* moves the string to our common/constants
* Simple function in your tests which gets either the configured env. or the default SIEM index.

```ts
const { DEFAULT_SIEM_INDEX } = '../../constants'
...
const getTestIndex = () => process.env.SIGNALS_INDEX || DEFAULT_SIEM_INDEX
...
expect(params.index).toEqual(getTestIndex());
```